submissions/xT0PERx/Friends app/js/index.js

```diff
@@ -0,0 +1,177 @@
+const friendsList = {
+  current: [],
+  changed: [],
+  count: 72,
+};
+
+const ALL_USERS_CARDS_FILTERS = friendsList.changed;
+const ALL_USERS_CARDS = friendsList.current;
+const USERS_LIST = document.getElementById("usersList");
+const FILTERS_MENU = document.getElementById("filters_menu");
+const NAME_ASCEND_BUTTON_FILTER = document.getElementById("name-ascend");
+const NAME_DESCEND_BUTTON_FILTER = document.getElementById("name-descend");
+const MALE_BUTTON_FILTER = document.getElementById("male");
+const FEMALE_BUTTON_FILTER = document.getElementById("female");
+const YOUNGER_FILTER_BUTTON = document.getElementById("younger-ftr");
+const SENIOR_FILTER_BUTTON = document.getElementById("senior-ftr");
+const RESET_BUTTON = document.getElementById("reset");
+const SEARCH_INPUT = document.querySelector(".searcher");
+
+function initApp() {
+  getUsers().then((users) => {
+    saveUsers(users);
+    createUsersCards(ALL_USERS_CARDS);
+  });
+}
+
+initApp();
+
+function getUsers() {
+  const url = `https://randomuser.me/api/1.3/?results=${friendsList.count}`;
+
+  return fetch(url)
+    .then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+    })
+    .then((data) => data.results)
+    .catch((err) => {
+      console.error(err.message);
+    });
+}
+
+function saveUsers(users) {
+  users.forEach(function (el) {
+    ALL_USERS_CARDS.push(el);
+    ALL_USERS_CARDS_FILTERS.push(el);
+  });
+}
+
+function createUsersCards(users) {
+  const lists = document.createDocumentFragment();
+
+  users.forEach(function (el) {
+    const card = document.createElement("li");
+    card.classList.add("userCard");
+    card.innerHTML = `<span class="name">${el.name.first} ${el.name.last}</span>
+    <img src="${el.picture.large}">
+    <span class="age">Age: ${el.dob.age}</span>
+    <span class="location">Location: ${el.location.city}</span>`;
+    lists.appendChild(card);
+  });
+
+  USERS_LIST.appendChild(lists);
+}
+
+FILTERS_MENU.addEventListener("click", function ({ target }) {
+  switch (target) {
+    case NAME_ASCEND_BUTTON_FILTER:
+      sortByAscendName();
+      break;
+    case NAME_DESCEND_BUTTON_FILTER:
+      sortByDescendName();
+      break;
+    case YOUNGER_FILTER_BUTTON:
+      sortAscend();
+      break;
+    case SENIOR_FILTER_BUTTON:
+      descendSort();
+      break;
+    case MALE_BUTTON_FILTER:
+      sortByMale();
+      break;
+    case FEMALE_BUTTON_FILTER:
+      sortByFemale();
+      break;
+    case RESET_BUTTON:
+      resetFilters();
+      break;
+  }
+});
```

Comment:
When you need to decide which sort of condition was selected, prefer value of input instead “check classname of pressed element”, or “compare pressed element to saved element” etc. You can have value female, get value and then just use it